explorer/interpreter/type_checker.cpp

```diff
@@ -5908,11 +5908,15 @@ auto TypeChecker::DeclareAliasDeclaration(Nonnull<AliasDeclaration*> alias,
            << "invalid target for alias declaration";
   }
 
-  CARBON_ASSIGN_OR_RETURN(Nonnull<const Value*> target,
-                          InterpExp(&alias->target(), arena_, trace_stream_));
-
   alias->set_static_type(&alias->target().static_type());
-  alias->set_constant_value(target);
+  // constant_value not needed for namespace alias because these are resolved by
+  // NameResolver
+  if (alias->target().static_type().kind() !=
+      Value::Kind::TypeOfNamespaceName) {
+    CARBON_ASSIGN_OR_RETURN(Nonnull<const Value*> target,
+                            InterpExp(&alias->target(), arena_, trace_stream_));
+    alias->set_constant_value(target);
+  }
```

Comment:
The special-casing in type_checker.cpp makes me wonder what would happen if you try using an aliased namespace in a place that's not expecting an aliased namespace. e.g., add a test for `namespace N; alias A = N; fn Main() -> i32 { return A; }` and see what happens.